weblate/trans/forms.py

```diff
@@ -2441,6 +2442,12 @@ def clean_zipfile(self):
         backup = ProjectBackup(fileio=zipfile)
         try:
             backup.validate()
+        except jsonschema.exceptions.ValidationError as error:
+            raise ValidationError(
+                gettext(
+                    "Could not load project backup: The backup is from an incompatible version. Please upgrade your Weblate instance."
+                )
```

Comment:
`backup.data["metadata"]["version"]` should contain Weblate version that did the backup; it might be useful to include it here.